Sources/NonEmpty/NonEmpty.swift

```diff
@@ -2,8 +2,6 @@
 public struct NonEmpty<Collection: Swift.Collection>: Swift.Collection {
   public typealias Element = Collection.Element
   public typealias Index = Collection.Index
-  public typealias Iterator = Collection.Iterator
-  public typealias SubSequence = Collection.SubSequence
```

Comment:
It seems it was a mistake to proxy the iterator and subsequence types through, and made mutations potentially unsafe, like in #40.